src/Components/Components/src/Routing/Router.cs

```diff
@@ -267,7 +267,7 @@ private void OnLocationChanged(object sender, LocationChangedEventArgs args)
             _locationAbsolute = args.Location;
             if (_renderHandle.IsInitialized && Routes != null)
             {
-                _ = RunOnNavigateAsync(NavigationManager.ToBaseRelativePath(_locationAbsolute), args.IsNavigationIntercepted);
+                _ = RunOnNavigateAsync(NavigationManager.ToBaseRelativePath(_locationAbsolute), args.IsNavigationIntercepted).Preserve();
```

Comment:
https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.valuetask.preserve?view=net-5.0 is as clear as mud😃 What's the TLDR for when `Preserve()` helps❔

Reply:
A `ValueTask<T>` can wrap one of three things: a `T`, a `Task<T>`, or an `IValueTaskSource<T>`.  There are a bunch of things we say you can't do with an arbitrary `ValueTask<T>` _because_ it might be wrapping an `IValueTaskSource<T>`... you can't block waiting for it to complete (because the IVTS implementation likely won't support that), you can't fire-and-forget it (because doing so likely will neglect to return a pooled object), you can't await it multiple times (because the first await completing is likely to return it to a pool for subsequent use), you can't use it concurrently (because that's the same as awaiting it multiple times), etc.  But in an implementation if you know you want to do one of these things, we provide Preserve as an escape hatch... basically if the implementation wraps an IValueTaskSource, it'll convert it to one that wraps a Task.  In a case like this line, that means if the implementation did wrap a pooled IVTS, we'll successfully return it back to the pool upon its completion.  If the ValueTask wrapped a T or a Task, Preserve is effectively a nop.